src/builders/indexable-post-builder.php

```diff
@@ -96,10 +97,11 @@ public function set_indexable_repository( Indexable_Repository $indexable_reposi
 	 * @return bool|Indexable The extended indexable. False when unable to build.
 	 *
 	 * @throws Post_Not_Found_Exception When the post could not be found.
+	 * @throws Post_Not_Built_Exception When the post should not be indexed.
 	 */
 	public function build( $post_id, $indexable ) {
 		if ( ! $this->post_helper->is_post_indexable( $post_id ) ) {
-			return false;
+			throw Post_Not_Built_Exception::because_not_viewable( $post_id );
```

Comment:
I think we should 
```
throw new Post_Not_Indexable_Exception();
```
and have a specific Exception for that case, with the relevant message.

Because now it throws a `not viewable` exception even though the culprit could be a different one, eg. `wp_is_post_revision`